graph.py

```diff
@@ -0,0 +1,71 @@
+from itertools import product
+import random
+from collections import deque
+
+class Vertex():
+    def __init__(self,name:str,value=None,children:list=None):
```

Comment:
If you're going to use type hints check out the `typing` module for the types that are less obvious. The aim should be to use the most restrictive type that allows all valid inputs.